app/cdash/include/CDash/System.php

```diff
@@ -0,0 +1,193 @@
+        $length = count($args);
+
+        foreach ($reversed as $arg) {
+            if (is_null($arg)) {
+                --$length;
+                continue;
+            }
+            break;
+        }
+
+        return array_splice($args, 0, $length);
+    }
+
+    /**
+     * @param $varname
+     * @param $newvalue
+     * @return void
+     */
+    public function ini_set($varname, $newvalue)
+    {
+        ini_set($varname, $newvalue);
+    }
+
+    /**
+     * @param null $name
+     * @return void
+     */
+    public function session_name($name = null)
+    {
+        self::call('session_name', [$name]);
+    }
+
+    /**
+     * @param null $cache_limiter
+     * @return void
+     */
+    public function session_cache_limiter($cache_limiter = null)
+    {
+        self::call('session_cache_limiter', [$cache_limiter]);
+    }
+
+    /**
+     * @param $lifetime
+     * @param $path
+     * @param $domain
+     * @param $secure
+     * @param $httponly
+     * @return void
+     */
+    public function session_set_cookie_params($lifetime)
+    {
+        session_set_cookie_params($lifetime);
+    }
+
+    /**
+     * @param array $options
+     * @return void
+     */
+    public function session_start($options = [])
+    {
+        self::call('session_start', $options);
+    }
+
+    /**
+     * @return void
+     */
+    public function session_regenerate_id()
+    {
+        session_regenerate_id();
+    }
+
+    /**
+     * @param null $id
+     * @return string
+     * @return void
+     */
+    public function session_id($id = null)
+    {
+        // This seems trivial, but it is definitely not
+        if (is_null($id)) {
+            return session_id();
+        } else {
+            return session_id($id);
+        }
+    }
+
+    /**
+     * @return void
+     */
+    public function session_destroy()
+    {
+        session_destroy();
+    }
+
+    /**
+     * @param $name
+     * @param $value
+     * @param $expire
+     * @param $path
+     * @param $domain
+     * @param $secure
+     * @param $httponly
+     * @return void
+     */
+    public function setcookie($name, $value, $expire, $path, $domain, $secure, $httponly)
+    {
+        setcookie($name, $value, $expire, $path, $domain, $secure, $httponly);
+    }
+
+    /**
+     * @return int
+     */
+    public function session_status()
+    {
+        return session_status();
+    }
+
+    /**
+     * @return void
+     */
+    public function session_write_close()
+    {
+        session_write_close();
+    }
+
+    /**
+     * @param $string
+     * @param bool $replace
+     * @param int|null $http_response_code
+     * @return void
+     */
+    public function header($string, $replace = true, $http_response_code = null)
+    {
+        header($string, $replace, $http_response_code);
+    }
+
+    /**
+     * @param $exit_message
+     * @return void
+     */
+    public function system_exit($exit_message = '')
+    {
+        // exit($exit_message);
```

Comment:
It looks like the only real place this gets called is `json_error_response`. When we call `json_error_response` the API endpoint stops executing. I'm worried that this will no longer the case now that `system_exit` is essentially a no-op.